File: python/백준 온라인 저지/2021.09.09/연결 요소의 개수.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10**10)

n, m = map(int, input().split())

graph = [[] for _ in range(n+1)]
for _ in range(m):
    s, e = map(int, input().split())
    graph[s].append(e)
    graph[e].append(s)

visited = [ False for _ in range(n+1) ]

def dfs(s):
    visited[s] = True
    for i in graph[s]:
        if not visited[i]:
            dfs(i)
cnt = 0
for i in range(1, n+1):
    if not visited[i]:
        cnt += 1
        dfs(i)

print(cnt)

File: Kotlin/Algorithm/src/연결 요소의 개수.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val (node, edge) = readLine().split(" ").map {it.toInt()}
    val graph = Array(node + 1) { ArrayList<Int>() }
    val visited = BooleanArray(node+1)
    var answer = 0

    repeat(edge) {
        val (s, e) = readLine().split(" ").map { it.toInt() }
        graph[s].add(e)
        graph[e].add(s)
    }

    fun dfs(now:Int) {
        visited[now] = true
        for (i in graph[now]) {
            when {
                !visited[i] -> dfs(i)
            }
        }
    }

    for (i in 1..node) {
        when {
            !visited[i] -> {
                dfs(i)
                answer ++
            }
        }
    }
    println(answer)
}